tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Shared/ContextPropagation.cs

```diff
@@ -17,13 +17,32 @@ internal static class ContextPropagation
     {
         internal const string InjectionKey = "_datadog";
 
-        private static void Inject(PropagationContext context, IDictionary messageAttributes, DataStreamsManager? dataStreamsManager, IMessageHeadersHelper messageHeadersHelper)
+        private static void Inject(PropagationContext context, IDictionary messageAttributes, DataStreamsManager? dataStreamsManager, IMessageHeadersHelper messageHeadersHelper, bool isLegacyHeaderNamesEnabled)
         {
             // Consolidate headers into one JSON object with <header_name>:<value>
             var sb = Util.StringBuilderCache.Acquire();
             sb.Append('{');
             SpanContextPropagator.Instance.Inject(context, sb, default(StringBuilderCarrierSetter));
 
+            if (dataStreamsManager != null && dataStreamsManager.IsEnabled && context.SpanContext?.PathwayContext != null)
+            {
+                var encodedBytes = PathwayContextEncoder.Encode(context.SpanContext.PathwayContext.Value);
+                var base64EncodedContext = Convert.ToBase64String(encodedBytes);
+                const string ddPathwayCtxBase64 = $"\"{DataStreamsPropagationHeaders.PropagationKeyBase64}\":\"";
+                sb.Append(ddPathwayCtxBase64)
+                  .Append(base64EncodedContext)
+                  .Append('"');
+
+                if (isLegacyHeaderNamesEnabled)
+                {
+                    // Both PropagationKeyBase64 and PropagationKey use the Base64 encoded context
+                    const string ddPathwayCtx = $"\"{DataStreamsPropagationHeaders.PropagationKey}\":\"";
+                    sb.Append(ddPathwayCtx)
```

Comment:
We need a comma here, but for some reason adding it in doesn't seem to get picked up locally 😖 

The content that this produces is `"dd-pathway-ctx-base64":"kjZ85G4hViqe65Xh6mSe65Xh6mQ=""dd-pathway-ctx":"kjZ85G4hViqe65Xh6mSe65Xh6mQ=","dd-pathway-ctx-base64":"a2paODVHNGhWaXFlNjVYaDZtU2U2NVhoNm1RPQ=="` 

note the missing comma and duplicated key?

Reply:
Looking into this at the moment

Reply:
I removed the JSON injection done here

1. It didn't properly handle `comma` values between the JSON so it produced Newtonsoft deserialization errors
2. It doesn't appear to be needed by my testing and was duplicating the headers.

And example of the JSONified headers now

```json
{
    "x-datadog-trace-id": "5758925048254194083",
    "x-datadog-parent-id": "9282207551708089276",
    "x-datadog-sampling-priority": "1",
    "x-datadog-tags": "_dd.p.dm=-0,_dd.p.tid=6741191500000000",
    "traceparent": "00-67411915000000004febcfd1bd4a61a3-80d10696cb024bbc-01",
    "tracestate": "dd=s:1;p:80d10696cb024bbc;t.dm:-0",
    "dd-pathway-ctx-base64": "R0lmbmJPT210ZUd3L2MvazZtU3cvYy9rNm1RPQ==",
    "dd-pathway-ctx": "GIfnbOOmteGw/c/k6mSw/c/k6mQ="
}
```

Reply:
Done in be18958a0b2707dc2bcaa68d436ae49287370ccb